lib/backend/k8s/conversion.go

```diff
@@ -230,7 +230,21 @@ func (c Converter) NetworkPolicyToPolicy(np *extensions.NetworkPolicy) (*model.K
 	// Generate the inbound rules list.
 	var inboundRules []model.Rule
 	for _, r := range np.Spec.Ingress {
-		inboundRules = append(inboundRules, c.k8sIngressRuleToCalico(r, np.ObjectMeta.Namespace)...)
+		inboundRules = append(inboundRules, c.k8sRuleToCalico(r.From, r.Ports, np.ObjectMeta.Namespace)...)
+	}
+
+	// Generate the outbound rules list.
+	var outboundRules []model.Rule
+	for _, r := range np.Spec.Egress {
+		outboundRules = append(outboundRules, c.k8sRuleToCalico(r.To, r.Ports, np.ObjectMeta.Namespace)...)
+	}
+
+	// Calculate Types setting.
```

Comment:
I think we need to use the exact values specified in `np.Spec.PolicyTypes` rather than calculating it here ourselves.

Reply:
D'oh yes, of course!